plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiConfig.java

```diff
@@ -0,0 +1,249 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.plugin.hudi;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.units.DataSize;
+
+import javax.validation.constraints.DecimalMax;
+import javax.validation.constraints.DecimalMin;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+
+import java.util.List;
+
+import static com.google.common.base.Strings.nullToEmpty;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+import static java.util.Locale.ENGLISH;
+import static org.apache.hudi.common.model.HoodieFileFormat.PARQUET;
+
+public class HudiConfig
+{
+    private static final Splitter COMMA_SPLITTER = Splitter.on(",").omitEmptyStrings().trimResults();
+
+    private String baseFileFormat = PARQUET.name();
+    private List<String> columnsToHide = ImmutableList.of();
+    private boolean metadataEnabled;
+    private boolean shouldSkipMetaStoreForPartition;
+    private boolean shouldUseParquetColumnNames = true;
+    private int partitionScannerParallelism = 16;
+    private int splitGeneratorParallelism = 16;
+    private int minPartitionBatchSize = 10;
+    private int maxPartitionBatchSize = 100;
+    private boolean sizeBasedSplitWeightsEnabled = true;
+    private DataSize standardSplitWeightSize = DataSize.of(128, MEGABYTE);
+    private double minimumAssignedSplitWeight = 0.05;
+    private int maxSplitsPerSecond = Integer.MAX_VALUE;
+    private int maxOutstandingSplits = 1000;
+
+    @NotNull
+    public String getBaseFileFormat()
+    {
+        return baseFileFormat;
+    }
+
+    @Config("hudi.base-file-format")
+    public HudiConfig setBaseFileFormat(String baseFileFormat)
+    {
+        this.baseFileFormat = baseFileFormat;
+        return this;
+    }
+
+    public List<String> getColumnsToHide()
+    {
+        return columnsToHide;
+    }
+
+    @Config("hudi.columns-to-hide")
+    public HudiConfig setColumnsToHide(String columnsToHide)
+    {
+        this.columnsToHide = COMMA_SPLITTER.splitToStream(nullToEmpty(columnsToHide))
+                .map(s -> s.toLowerCase(ENGLISH))
+                .collect(toImmutableList());
+        return this;
+    }
+
+    @Config("hudi.metadata-enabled")
+    @ConfigDescription("Fetch the list of file names and sizes from metadata rather than storage")
```

Comment:
Hudi has an internal metadata table that is consistent with the data table. When this config is enabled, then instead of actual recursive file listing, Hudi will use the metadata table to speed up split generation. More details at https://hudi.apache.org/docs/metadata